### PR TITLE
Gvt changes in preperation for changes to Mount device only after OS has discovered it.

### PR DESCRIPTION
**Pull request by kerneltime**
The following changes are part of this PR.

1. Add new make gvt target that drops user into a container in which the user can add vendor sw. Documentation for gvt updated.
2. Many trailing white spaces removed.

Testing:

Each run command part of CI will run this code.

### CODE REVIEW

File: vmdk_plugin/plugin.go [hunk 1]
```diff
@@ -117,19 +116,53 @@ func (d *vmdkDriver) mountVolume(name string) (string, error) {
...
+	}
+
+	watcher, err := inotify.NewWatcher()
+	if err != nil {
+		log.WithFields(
+			log.Fields{"name": name, "dir": mountpoint},
+		).Error("Failed to create watcher ")
+		return mountpoint, err
+	}
+	err = watcher.Watch("/dev/disk/by-path")
```
Comment (govint) [inline vmdk_plugin/plugin.go]: Is it possible to watch the specific device (do the watch on the device thats to be mounted. Do it on the device returned by get_device_path(). Else will the watcher return if some device gets created, not necessarily the one for the attach below.

**Comment by govint** [comment]
Could add a test where multiple (say 10 - 20) containers are started in parallel and each of those are able to mount their volumes successfully.

File: vmdk_plugin/plugin.go (hunk 1 above)
Comment (kerneltime) [inline vmdk_plugin/plugin.go]: The select loop filters out devices we do not care about.

File: vmdk_plugin/plugin.go [hunk 2]
```diff
@@ -117,19 +116,53 @@ func (d *vmdkDriver) mountVolume(name string) (string, error) {
...
+	select {
+	case ev := <-watcher.Event:
+		log.Debug("event: ", ev)
+		if ev.Name == device {
+			// Log when the device is discovered
+			log.WithFields(
+				log.Fields{"name": name, "event": ev},
+			).Info("Attach complete ")
+			break
+		}
```
Comment (msterin) [inline vmdk_plugin/plugin.go]: timeout is needed here

File: vmdk_plugin/plugin.go [hunk 3]
```diff
@@ -117,19 +116,53 @@ func (d *vmdkDriver) mountVolume(name string) (string, error) {
...
+	if d.useMockEsx {
+		return mountpoint, fmt.Errorf("No device to mount.")
+	}
+
+	watcher, err := inotify.NewWatcher()
+	if err != nil {
+		log.WithFields(
+			log.Fields{"name": name, "dir": mountpoint},
+		).Error("Failed to create watcher ")
+		return mountpoint, err
```
Comment (pdhamdhere) [inline vmdk_plugin/plugin.go]: Watcher errors should not result in mountVolume error. We should log & proceed to check if disk is mounted anyway.

File: vmdk_plugin/plugin.go [hunk 4]
```diff
@@ -117,19 +116,53 @@ func (d *vmdkDriver) mountVolume(name string) (string, error) {
...
+	}
+	err = watcher.Watch("/dev/disk/by-path")
+	if err != nil {
+		log.WithFields(
+			log.Fields{"name": name, "dir": mountpoint},
+		).Error("Failed to watch /dev ")
+		return mountpoint, err
```
Comment (pdhamdhere) [inline vmdk_plugin/plugin.go]: Same here. Let's log & proceed.

File: vmdk_plugin/plugin.go (hunk 2 above)
Comment (pdhamdhere) [inline vmdk_plugin/plugin.go]: +1 on timeout

File: vmdk_plugin/plugin.go [hunk 5]
```diff
@@ -117,19 +116,53 @@ func (d *vmdkDriver) mountVolume(name string) (string, error) {
...
+			// Log when the device is discovered
+			log.WithFields(
+				log.Fields{"name": name, "event": ev},
+			).Info("Attach complete ")
+			break
+		}
+	case err := <-watcher.Error:
+		log.WithFields(
+			log.Fields{"name": name, "dir": mountpoint, "error": err},
+		).Error("Hit error during watch")
```
Comment (pdhamdhere) [inline vmdk_plugin/plugin.go]: Same here. We should ignore any errors reported by watcher and proceed optimistically.

**Comment by kerneltime** [comment]
Reducing this PR to be just gvt.

**Comment by msterin** [comment]
LGTM , just please update the PR description

**Pull request merged**